src/pymatgen/core/structure.py

```diff
@@ -4746,7 +4746,7 @@ def merge_sites(self, tol: float = 0.01, mode: Literal["sum", "delete", "average
                 offset = self[i].frac_coords - coords
                 coords += ((offset - np.round(offset)) / (n + 2)).astype(coords.dtype)
                 for key in props:
-                    if props[key] is not None and self[i].properties[key] != props[key]:
```

Comment:
Using `array_equal` may not be a good idea either as the property could be (sequence of) floats, the test failure should be related to this: https://github.com/materialsproject/pymatgen/blob/31f1e1fb8cc1bb517d59ea8e484965bb641c42a0/tests/core/test_structure.py#L1667-L1681

Reply:
Property can actually be anything, including value supplied by user.